### ISSUE TITLE
PC12 compatibility?

### ISSUE THREAD

**Issue by arphex** [privatepublic-de/faderfox-editor#6]
would help a lot.... many knobs...

**Comment by victorcampos** [comment]
@privatepublic-de I'm also really interested on having PC12 compatibility and can help with it, been a software engineer for 15+ years.

As you had help from the people at Faderfox themselves for the EC4/PC4 editors, do you think I could start working on this by adapting the PC4 editor SysEx to the PC12? Or do you know if Faderfox have a different SysEx implementation between different lines of controllers which could make this trickier than my initial impression?

Cheers!

**Comment by arphex** [comment]
we can try asking this from faderfox, they have a contact email here http://www.faderfox.de/contact.html

**Comment by privatepublic-de** [comment]
Yeah, exactly, please contact Mathias from Faderfox. He's nice and cooperative. All the SysEx docs in this repository (.txt files in /doc folder) are from him. I'd expect that the PC12 SysEx-format builds on the PC4's with extensions to the additional buttons, the high-res encoder and controller groups.

**Comment by arphex** [comment]
> Yeah, exactly, please contact Mathias from Faderfox. He's nice and cooperative. All the SysEx docs in this repository (.txt files in /doc folder) are from him. I'd expect that the PC12 SysEx-format builds on the PC4's with extensions to the additional buttons, the high-res encoder and controller groups.

Thanks! I have sent an email but no response yet

**Comment by privatepublic-de** [comment]
If you would like to build your PC12 version on this (actually quite dated) repository I can invite you as collaborator. But feel free to fork if you'd rather do that. Unfortunable I don't have a Faderfox PC12, so I wont be able to test the new stuff.

**Comment by arphex** [comment]
I have the PC12 so I can try out any code anyone develops and get back with results.   However perhaps important to get the sysex specs first, as reverse engineering hex code might not be the optimal thing to do - or doable in that respect

**Comment by tranzfuse** [comment]
Hello from the future. Was hoping to find a PC12 web editor and stumbled upon this repo. Any momentum on a PC12 editor implementation at this point? Thanks all for your efforts!

**Comment by mattpstv** [comment]
i believe a PC12/MX12 web editor solution based on the brilliant efforts of @privatepublic-de is unfortunately stuck somewhere in the Bonnie Tyler-stage of development (ie. "holding out for a hero")

**Comment by tranzfuse** [comment]
I was considering taking this on as a side project to refresh my web development skills so I reached out to faderfox to check in. Mathias (faderfox) replied and says:

> unfortunately I have no documentation about the sysex format of PC12/MX12 available currently.
> it’s not possible to work on this without the exact documentation of the memory locations etc.
> I’m very busy with production of my current models.

Sounds like we will have to manually program our devices in the meantime...

**Comment by arphex** [comment]
i tried looking at the sysex a while ago w some reverse engineering, similarities exist with the other models that have editors but I guess Mathias is right, without knowing the exact locations etc. might not be possible - which is a real shame.

**Comment by arphex** [comment]
> I was considering taking this on as a side project to refresh my web development skills so I reached out to faderfox to check in. Mathias (faderfox) replied and says:
> 
> > unfortunately I have no documentation about the sysex format of PC12/MX12 available currently.
> > it’s not possible to work on this without the exact documentation of the memory locations etc.
> > I’m very busy with production of my current models.
> 
> Sounds like we will have to manually program our devices in the meantime...

manually programming is fine, but recalling presets, remembering where you stored something a while ago - thats all too cumbersome. Would be great to get the specs at some point - its an expensive and quality device which can only get better with some decent software!

**Comment by notno** [comment]
This is a real shame. I hope he finds the time to support the community of people who use his products.
